src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!prey.hidden && prey instanceof Herbivore) {
+      prey.health -= 50;
+    }
```

Comment:
The check for `prey.hidden` and `instanceof Herbivore` is correctly done in a single if statement. However, consider checking if the prey's health after being bitten is less than or equal to 0 within this if block, before filtering the `Animal.alive` array, to avoid unnecessary filtering when the prey is not a Herbivore or is hidden.